databricks-sdk-java/src/main/java/com/databricks/sdk/core/retry/IdempotentRequestRetryStrategy.java

```diff
@@ -0,0 +1,17 @@
+package com.databricks.sdk.core.retry;
+
+import com.databricks.sdk.core.DatabricksError;
+import com.databricks.sdk.core.http.Response;
+
+public class IdempotentRequestRetryStrategy implements RetryStrategy {
+  @Override
+  public boolean isRetriable(Response response, Exception e) {
+    // TODO: Not all errors are retriable, we can skip retrying in certain cases
```

Comment:
This is definitely not right. We hit this codepath when actual 4xx errors are returned that mustn't be retried in the client (e.g. Bad Request, Resource Not Found, etc.). 

I think there are 4 classes of errors here:
1. The server definitely rejected the request with a non-retriable error.
2. The server possibly received the request, but there was a timeout, either client-side or server-side, that prevented the request from being handled.
3. The server rejected the request with a retriable error. Today, this includes 429, 501 and 503 and "transient errors" below.
4. The server definitely did not receive the request.

I think IdempotentRetryStrategy retries on 2, 3, and 4, and NonIdempotentRetryStrategy retries on 3 and 4 only.

Reply:
I added it to IdempotentRequestRetryStrategy as a do not retry if 1 which is opposite to our approach for NonIdempotentRequestRetryStrategy i.e. retry on 3, 4. It seems a bit counter-intuitive at a high level but seems like the right thing to do? Let me know your thoughts